src/easydynamics/utils/convolution.py

```diff
@@ -0,0 +1,597 @@
+    if (
+        isinstance(sample_component, Gaussian)
+        and isinstance(resolution_component, Lorentzian)
+    ) or (
+        isinstance(sample_component, Lorentzian)
+        and isinstance(resolution_component, Gaussian)
+    ):
+        if isinstance(sample_component, Gaussian):
+            G, L = sample_component, resolution_component
+        else:
+            G, L = resolution_component, sample_component
+        center = (G.center.value + L.center.value) + off
+        area = G.area.value * L.area.value
+        return True, _voigt_eval(x, center, G.width.value, L.width.value, area)
+
+    return False, np.zeros_like(x, dtype=float)
+
+
+def _gaussian_eval(
+    x: np.ndarray, center: float, width: float, area: float
+) -> np.ndarray:
+    """
+    Evaluate a Gaussian function. y = (area / (sqrt(2pi) * width)) * exp(-0.5 * ((x - center) / width)^2)
+    All checks are handled in the calling function.
+
+    args:
+        x : np.ndarray
+            1D array of x values where the Gaussian is evaluated.
+        center : float
+            The center of the Gaussian.
+        width : float
+            The width (sigma) of the Gaussian.
+        area : float
+            The area under the Gaussian curve.
+    Returns:
+        np.ndarray
+            The evaluated Gaussian values at x.
+    """
+    return (
+        area
+        * 1
+        / (np.sqrt(2 * np.pi) * width)
+        * np.exp(-0.5 * ((x - center) / width) ** 2)
+    )
+
+
+def _lorentzian_eval(
+    x: np.ndarray, center: float, width: float, area: float
+) -> np.ndarray:
+    """
+    Evaluate a Lorentzian function. y = (area * width / pi) / ((x - center)^2 + width^2).
+    All checks are handled in the calling function.
+
+    args:
+        x : np.ndarray
+            1D array of x values where the Lorentzian is evaluated.
+        center : float
+            The center of the Lorentzian.
+        width : float
+            The width (HWHM) of the Lorentzian.
+        area : float
+            The area under the Lorentzian.
+    Returns:
+        np.ndarray
+            The evaluated Lorentzian values at x.
+    """
+    return area * width / np.pi / ((x - center) ** 2 + width**2)
+
+
+def _voigt_eval(
+    x: np.ndarray, center: float, g_width: float, l_width: float, area: float
+) -> np.ndarray:
+    """
+    Evaluate a Voigt profile function using scipy's voigt_profile.
+    args:
+        x : np.ndarray
+            1D array of x values where the Voigt profile is evaluated.
+        center : float
+            The center of the Voigt profile.
+        g_width : float
+            The Gaussian width (sigma) of the Voigt profile.
+        l_width : float
+            The Lorentzian width (HWHM) of the Voigt profile.
+        area : float
+            The area under the Voigt profile.
+    Returns:
+        np.ndarray
+            The evaluated Voigt profile values at x.
+    """
+
+    return area * voigt_profile(x - center, g_width, l_width)
```

Comment:
scipy's `voigt_profile` is normalized, so multiplying by `area` (which is the product of the two component areas) should work. However, this needs verification with unit tests. Please add some.